test/Datadog.Trace.TestHelpers/TcpPortProvider.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.NetworkInformation;
+
+namespace Datadog.Trace.TestHelpers
+{
+    /// <summary>
+    /// Helper class that tries to provide unique ports numbers across processes and threads in the same machine.
+    /// Used avoid port conflicts in concurrent tests that use the Agent, IIS, HttpListener, HttpClient, etc.
+    /// This class cannot guarantee a port is actually available, but should help avoid most conflicts.
+    /// </summary>
+    public static class TcpPortProvider
+    {
+        private static readonly ConcurrentBag<int> ReturnedPorts = new ConcurrentBag<int>();
+
+        // multiply by 100 so each process gets a built-in 100-port buffer
+        private static readonly int MinPort = 10000 + ((Process.GetCurrentProcess().Id * 100) % 10000);
+
+        public static int GetOpenPort(int minPort = 0, int maxPort = ushort.MaxValue)
```

Comment:
Since the `minPort` and `maxPort` parameters are not used, I'd say we can remove them.

Reply:
Good point. Removed in 7c0528b886902a751903b15379b3eff358e1c168.